cs/remote/src/FASTER.common/SimpleObjectPool.cs

```diff
@@ -12,20 +11,26 @@ namespace FASTER.common
     /// Object pool
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public class SimpleObjectPool<T> : IDisposable where T : class, IDisposable
+    public class SimpleObjectPool<T> : IDisposable where T : class
     {
         private readonly Func<T> factory;
+        private readonly Action<T> destructor;
```

Comment:
why this change? there is other code we have that depends on this.

Reply:
Needed to accommodate objects (e.g., SEAAs) that needs custom Dispose() logic. I have changed the default argument ordering so it should now be a backward compatible API change for existing code.